server/startup/accounts.js

```diff
@@ -105,7 +128,11 @@ export default function () {
       if (!user.services) {
         roles[shopId] = shop.defaultVisitorRole || defaultVisitorRole;
       } else {
```

Comment:
I would prefer you use `else if (isVendor)` to avoid so many nesting. You can simplify further